DSharpPlus.Commands/Processors/SlashCommands/SlashCommandProcessor.cs

```diff
@@ -450,17 +450,37 @@ private async Task<DiscordApplicationCommandOption> ToApplicationParameterAsync(
         }
 
         Dictionary<CommandParameter, object?> parsedArguments = [];
+        CommandParameter? autoCompleteParameter = null;
+        DiscordInteractionDataOption? autoCompleteOption = null;
         try
         {
             // Parse until we find the parameter that the user is currently typing
-            while (converterContext.NextParameter() && !converterContext.Options.ElementAt(converterContext.ParameterIndex).Focused)
+            while (converterContext.NextParameter())
             {
+                DiscordInteractionDataOption? option = converterContext.Options.FirstOrDefault(x => x.Name.Equals(converterContext.Parameter.Name, StringComparison.OrdinalIgnoreCase));
```

Comment:
We shouldn't ignore the case here. While we can right now due to all parameters being mutated to `snake_case`, this behavior will not stay forever. Instead I'd use `StringComparison.Ordinal`.

Reply:
I've had an error on a command because i had a capitalize param name on it, that's why i made it IgnoreCase, maybe I can add a TODO here to specify to switching this comparaison to Ordinal as soon as the snake_case mutating thing is gone ?

Reply:
Yeah that'll work